README.md

```diff
@@ -1,3 +1,126 @@
+             |                  /      |                      |
+        +----|----+            /  +----|----+            +----|----+
+        |    v    |           /   |    v    |            |    v    |
+    UO <==       <== DI <==\ / UO<==       <== NI    UO <==       <== NI
+        |    f    |         X     |    g    |     =      | f |>| g |
+    UI ==>       ==> DO ===/ \ UI==>       ==> NO    UI ==>       ==> NI
+        |    |    |           \   |    |    |            |    |    |
+        +----|----+            \  +----|----+            +----|----+
+             v                  \      v                      v
+             FR                  \====>DI                     FR
+```
+
+If that all seems complicated, don't worry!  The diagrams are always accompanied
+by proper explanations of the semantics of their respective methods in *plain
+english*.  If the documentation is ever unclear, feel free to ask a question,
+file an issue, or put it in your own words in a pull request.  
+
+# Programming With Pipes
+
+By now you may be wondering how one actually goes about using `Aquifer` to
+produce useful programs.  Returning to the `Proxy` type from earlier, each of
+its 5 parameters is a different unhandled part of the flow of data through
+a pipe that you the user must "seal" or "fuse" using the other combinators in
+this library.  For example, the operation `yield` can be thought of as:
+
+```swift
+/// Yield leaves a hole of type A that we need to fuse somehow.
+func yield<A>(value : A) -> Producer<A, ()>.T
+```
+
+`yield` is an operation that returns a value with a "hole" in it that we need to
+fill.  In fact, if we were simply to use `yield` by itself, our program would
+block indefinitely waiting for the unfused hole to be filled!  A `yield` is
+usually connected to, naturally, a `Consumer` pipe, which itself has a hole of
+type `A`.  Here we'll hook up the function `stdinLn` (which returns
+a `Producer<String, ()>` that `yield`s values from stdin) to the function
+`stdoutLn` (which returns a `Consumer<String, ()>` that `await`s values from
+some `Producer` and prints them to stdout) with one of many fusing operators `>->`
+
+```swift
+let pipe = stdinLn() >-> stdoutLn()
+```
+
+The thing to notice about `pipe` is that Swift believes it has the type
+
+```
+Proxy<X, (), (), X, ()>
+```
+
+By consulting the handy table of `typealias`es in [`Proxy`](Aquifer/Proxy.swift)'s
+definition file it's easy to see that this corresponds to the `Effect` alias.  An 
+`Effect` is *only* produced when every input and output from all parts of the pipe
+have been fused away, leaving only a program that can be run with the `runEffect` 
+combinator.
+
+Putting it all together yields
+
+```swift
+/// This program acts like `cat` by blitting any input from stdin to stdout.
+runEffect <| stdinLn() >-> stdoutLn()
+```
+
+# System Requirements
+
+Aquifer supports OS X 10.9+ and iOS 7.0+.
+
+# Further Reading
+
```

Comment:
No idea.  We could point them towards to original pipes tutorial, but I don't know how helpful it would be.